timeseries/src/autogluon/timeseries/models/ensemble/basic.py

```diff
@@ -62,11 +65,21 @@ def _fit(
         assert model_scores is not None
 
         weight_scheme = self.get_hyperparameters()["weight_scheme"]
-        self.model_to_weight = {}
-        score_transform = np.square if weight_scheme == "sq" else (lambda x: np.exp(np.clip(x, a_min=0, a_max=70)))
-
-        for model_name in predictions_per_window.keys():
-            self.model_to_weight[model_name] = score_transform(1.0 / (-model_scores[model_name] + 1e-5))
 
+        # drop NaNs
+        model_scores = {k: v for k, v in model_scores.items() if np.isfinite(v)}
+        assert len(model_scores) > 0, (
+            "All models have NaN scores. At least one model must score successfully to fit an ensemble"
+        )
+
+        score_transform = {
+            "sq": lambda x: np.square(np.reciprocal(x)),
+            "inv": lambda x: np.reciprocal(x),
+            "sqrt": lambda x: np.sqrt(np.reciprocal(x)),
+        }[weight_scheme]
+
+        self.model_to_weight = {
+            model_name: score_transform(-model_scores[model_name] + 1e-5) for model_name in model_scores.keys()
```

Comment:
This line assumes that all model scores are negative (otherwise `np.sqrt` will be undefined). This currently happens to be true, but is not guaranteed in general. Maybe add an assertion above?